debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java

```diff
@@ -137,7 +137,7 @@ public void execute(ChangeEventSourceContext context) throws InterruptedExceptio
                 try {
                     connection.getChangesForTables(tablesSlot.get(), fromLsn, currentMaxLsn, resultSets -> {
 
-                        int operationOrderInInitialtx = 1;
+                        long eventSerialNoInInitialTx = 0;
```

Comment:
@jpechane, I've changed this to be consistent with the case where the value is not available after restarting the connector from an old offset version without that field. Does this make sense?

Reply:
@gunnarmorling I think you've desynced the counter in this case. The serialNo is counted from 1 and you are here counting it from 0 so I've a strong feeling you'll either skip or reprocess one event.
The zero in snapshot was chosen deliberately to make sure that all events with the same LSNs are replayed. If we'd set it to 1 it would mean that the first event was processed but we do not if first or second or another event was processed, hence I am playing it safe.

Reply:
Ok, I wasn't quite sure :) I'll revert this one then.

Reply:
Updated and force-pushed.

Reply:
@gunnarmorling Apart from this change when the tests pass the PR can be merged.